### PR TITLE
21092 Dissolutions Job - Add meta data and unit tests to stage 1 

### PR DESCRIPTION
**Pull request by leodube-aot**
*Issue #:* /bcgov/entity#21092

*Description of changes:*
- Added meta data to created batch
- Added unit tests 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: jobs/involuntary-dissolutions/config.py
Comment (leodube-aot) [inline jobs/involuntary-dissolutions/config.py:83]: Changing these defaults to `0` makes the logic cleaner on lines 115-116 in `involuntary_dissolutions.py`, however if we want to keep it as `None` I can just add checks in that file before creating the `timedelta`. Note that for some reason `app.config.get('STAGE_2_DELAY', 0)` will return `None` if the fallback here is `None`

File: jobs/involuntary-dissolutions/Makefile
Comment (leodube-aot) [inline jobs/involuntary-dissolutions/Makefile:42]: Upgrading pip to latest 20.x.x version seems to not break the unit tests and also installs properly for me. It does however appear to throw lots of deprecation warnings when running the unit tests. I am not sure if this is directly related to the pip version, or if it is because I am manually installing some `legal_api` dependencies that are not getting properly installed like `nest_asyncio`.

Screenshot 2024-06-05 at 7 47 47 AM

File: jobs/involuntary-dissolutions/involuntary_dissolutions.py
Comment (chenhongjing) [reply jobs/involuntary-dissolutions/involuntary_dissolutions.py:82]: I've tried a bit, this also works.
```python
tz = pytz.timezone('US/Pacific')
today_date = tz.localize(datetime.today()).date()

batch_today = (
    db.session.query(Batch)
    .filter(Batch.batch_type == Batch.BatchType.INVOLUNTARY_DISSOLUTION)
    .filter(cast(Batch.start_date, Date) == today_date)
).all()
```

Comment (leodube-aot) [reply jobs/involuntary-dissolutions/involuntary_dissolutions.py]: I like this approach more, updated to match!

File: jobs/involuntary-dissolutions/tests/unit/test_involuntary_dissolutions.py
Comment (argush3) [inline jobs/involuntary-dissolutions/tests/unit/test_involuntary_dissolutions.py:34]: typo

**Review by argush3 — APPROVED** [review]
Looks good!  Thanks for the tests! 

Pre-approving with assumption linting and testing errors are resolved.

File: jobs/involuntary-dissolutions/config.py
Comment (chenhongjing) [reply jobs/involuntary-dissolutions/config.py]: I prefer keeping it as `None` so in code of stage 2, it will skip the run if such config value doesn't exist. This makes more sense to me. 🤔

Comment (leodube-aot) [reply jobs/involuntary-dissolutions/config.py]: Sounds good, updated!

**Review by chenhongjing — APPROVED** [review]
LGTM 😸  I'll rebase my code after you merge it.

**Pull request merged**
